### PR TITLE
Add de.peterge.combined

### PR DESCRIPTION
**Pull request by peterge1998**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
you should make sure it's only built for x86_64. See https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

**Review by hfiguiere — CHANGES_REQUESTED** [review]
you are missing an appdata file.

File: de.peterge.combined.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+app-id: de.peterge.combined
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: "Combined!.x86_64"
+
+finish-args:
+  - --socket=x11
```
Comment (hfiguiere) [inline de.peterge.combined.yml:8]: you need `- --share=ipc` for X11. And I suspect you also want `- --device=dri` for openGL.

File: de.peterge.combined.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+modules:
+  - name: combined
+    buildsystem: simple
+    build-commands:
+      - install -D "Combined!.x86_64" "/app/bin/Combined!.x86_64"
+      - install -D "Combined!.pck" "/app/bin/Combined!.pck"
+      - install -Dm644 icon.png -t /app/bin/icon.png
+      - install -Dm644 icon.png -t /app/share/icons/hicolor/256x256/apps/
+      - install -Dm644 de.peterge.combined.desktop /app/share/applications/de.peterge.comdined.desktop
```
[the remark this answers is not in the document]
Comment (peterge1998) [reply de.peterge.combined.yml:18]: I don't get this. What is correct here?
`      - install -Dm644 icon.png -t /app/bin/`

**Comment by peterge1998** [comment]
> you should make sure it's only built for x86_64. See https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64
Added a basic flathub.json in 54d9ade2b29eaa5bc99529400aa3f0f2e5f646a9.

>You need a rename-icon: icon
Added rename-icon: icon in e8682113ad979f496cc6f8825ea0836b0b0eb5a7.

And the OpenGL/x11 lines in e9c6287e053c894e32ce6525715bb07e3b7e6348.

Sorry, but I am new to publishing on Flathub ^^

File: de.peterge.combined.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply de.peterge.combined.yml:21]: definitely not.

the `.desktop` file has `Icon=icon.png` and that's wrong.  It should be `Icon=icon` (and `rename-icon` take care of the rest), to match with what you installed.

**Comment by hfiguiere** [comment]
still missing the appdata file.

**Comment by nedrichards** [comment]
bot, build de.peterge.combined

(this will fail, as @hfiguiere says, but it'll help you see the process and you can call the command when you push a new commit to check it)

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
Queued test build for de.peterge.combined.

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
What is going wrong here? [The `.desktop` file](https://gitlab.com/peterge/combined-assets/-/blob/main/de.peterge.combined.desktop)

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

File: de.peterge.combined.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hfiguiere) [inline de.peterge.combined.metainfo.xml:2]: this file isn't installed. hence the build failure.

File: de.peterge.combined.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+<component type="game">
+  <id>com.valvesoftware.SteamLink</id>
+  <launchable type="desktop-id">de.peterge.combined</launchable>
+  <metadata_license>MIT</metadata_license>
```
Comment (hfiguiere) [inline de.peterge.combined.metainfo.xml:6]: we prefer CC0-1.0 for the metadata license.

File: de.peterge.combined.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="game">
+  <id>com.valvesoftware.SteamLink</id>
```
Comment (hfiguiere) [inline de.peterge.combined.metainfo.xml:4]: the ID doesn't match.

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

File: de.peterge.combined.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+modules:
+  - name: combined
+    buildsystem: simple
+    build-commands:
+      - install -D "Combined!.x86_64" "/app/bin/Combined!.x86_64"
+      - install -D "Combined!.pck" "/app/bin/Combined!.pck"
+      - install -Dm644 icon.png -t /app/bin/icon.png
+      - install -Dm644 icon.png -t /app/share/icons/hicolor/256x256/apps/
+      - install -Dm644 de.peterge.combined.desktop /app/share/applications/de.peterge.combined.desktop
+      - install -Dm644 de.peterge.combined.metainfo.xml /app/share/de.peterge.combined.metainfo.xml
```
Comment (hfiguiere) [inline de.peterge.combined.yml:22]:
```suggestion
      - install -Dm644 de.peterge.combined.metainfo.xml /app/share/metainfo/de.peterge.combined.metainfo.xml
```

Comment (hfiguiere) [reply de.peterge.combined.yml:22]: but the file isn't listed below in the `sources` so it won't work until it is.

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by hfiguiere** [comment]
you can't install a file that's not listed in the sources.

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
image
Okay, finally the page in gnome's software looks good. One last thing: it does not display my screenshot, but maybe this does not work for the beta build yet. Finally ready to become offical available on flathub :rocket:

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Comment by peterge1998** [comment]
I did not notice this error, which prevents the app from running, on a wayland session yet: `initialize: X11 Display is not available`
Is there a way how i can run my app on wayland? If i run the .x86_64 file on a wayland session directly, it starts fine. What is the best way do define it with `--socket=fallback-x11`?

**Comment by peterge1998** [comment]
bot, build de.peterge.combined

**Pull request closed without merging**